test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -809,6 +848,14 @@ func runWorkload(b *testing.B, tc *testCase, w *workload) []DataItem {
 				}
 			}
 
+		case *sleepOp:
+			timer := time.NewTimer(concreteOp.Duration)
+			select {
+			case <-ctx.Done():
+				return nil
```

Comment:
why return here?

Reply:
I can't recall exactly what I was thinking a year ago, but it looks like the test needs to be terminated here because `case <-ctx.Done()` means it timed out.
I will add an error log here to inform users that a test is finished by timeout.

Reply:
I'm fine with leaving empty here, as in the upper `for` loop, we have to logic to log the ctx.Err():

```go
case <-ctx.Done():
			b.Fatalf("op %d: %v", opIndex, ctx.Err())
```

PS: I _guess_  the `return` was initially leveraged from other code here, but whose logic is "return"ing from a goroutine.

Reply:
> I'm fine with leaving it empty here, as in the upper for loop, we have to logic to log the ctx.Err():

make sense. Fixed as your suggestion.